dmoj/cptbox/ptproc.cpp

```diff
@@ -232,21 +235,28 @@ int pt_process::monitor() {
                             case PTBOX_HANDLER_CALLBACK:
                                 if (callback(context, syscall))
                                     break;
-                                //printf("Killed by callback: %d\n", syscall);
+                                // printf("Killed by callback: %d\n", syscall);
                                 exit_reason = protection_fault(syscall);
                                 continue;
                             default:
                                 // Default is to kill, safety first.
-                                //printf("Killed by DISALLOW or None: %d\n", syscall);
+                                // printf("Killed by DISALLOW or None: %d\n", syscall);
                                 exit_reason = protection_fault(syscall);
                                 continue;
                         }
+                    } else if (!callback(context, syscall)) {
```

Comment:
Might want a comment here that this is specifically for out-of-range syscalls on ARM platforms, probably linking https://github.com/DMOJ/judge/issues/128 for good measure.

Reply:
It's not just ARM, but it's there to catch other stupidity.